app/controllers/hub/state_file/efile_submissions_controller.rb

```diff
@@ -81,7 +81,13 @@ def transition_to
         authorize! :update, @efile_submission
         metadata = { initiated_by_id: current_user.id }
         if to_state == "rejected"
-          metadata[:error_code] = EfileError.where(service_type: "state_file_#{@efile_submission.data_source.state_code}").last.code
+          if params[:auto_cancel]
+            metadata[:error_code] = EfileError.where(service_type: "state_file_#{@efile_submission.data_source.state_code}", auto_cancel: true).last.code
+          elsif params[:auto_wait]
+            metadata[:error_code] = EfileError.where(service_type: "state_file_#{@efile_submission.data_source.state_code}", auto_wait: true).last.code
+          else
+            metadata[:error_code] = EfileError.where(service_type: "state_file_#{@efile_submission.data_source.state_code}", auto_cancel: false, auto_wait: false).last.code
+          end
```

Comment:
I don't really understand `auto_cancel` and `auto_wait`, but I assume these are different from the "regular rejection" which means these don't cancel and these don't wait -- they just _sit_ in rejected state/limbo and need manual handling?

Reply:
yeah basically `auto_cancel` = we know we cannot help the client and they need to file another way and `auto_wait` = the client needs to fix something so we can give them the go-ahead to do that. the goal is for the majority of our errors to go into one of these categories because everything else needs to be investigated by our team